### PR TITLE
perf: added http and grpc server in diff goroutines

### PR DESCRIPTION
**Pull request by Sanskarzz**
```
sanskar@sanskar-HP-Laptop-15s-du1xxx:~/kyverno-envoy-plugin$ go run main.go 
Starting HTTP server on Port 8080
Starting GRPC server on Port 9090
```

### CODE REVIEW

**Comment by eddycharly** [comment]
A couple of things to add, graceful shutdown and system events (Ctrl+C)

**Comment by Sanskarzz** [comment]
After hitting `Ctrl-C` the program will end in 5 second .

File: main.go [hunk 1]
```diff
@@ -34,24 +50,41 @@ func handler(w http.ResponseWriter, r *http.Request) {
...
+	grpcCtx, grpcCancel := context.WithCancel(context.Background())
 
 	srv := NewServers()
-	go srv.startHTTPServer()
-	go srv.startGRPCServer()
-	select {}
+	go srv.startHTTPServer(httpCtx)
+	go srv.startGRPCServer(grpcCtx)
+	done := make(chan os.Signal, 1)
+	signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
```
Comment (eddycharly) [inline main.go:83]: use a context instead of chan

File: main.go [hunk 2]
```diff
@@ -34,24 +50,41 @@ func handler(w http.ResponseWriter, r *http.Request) {
...
+
+	<-done
+	httpCancel()
+	grpcCancel()
+	time.Sleep(5 * time.Second)
```
Comment (eddycharly) [inline main.go:88]: this is not graceful

File: main.go [hunk 3]
```diff
@@ -34,24 +50,41 @@ func handler(w http.ResponseWriter, r *http.Request) {
...
-	go srv.startGRPCServer()
-	select {}
+	go srv.startHTTPServer(httpCtx)
+	go srv.startGRPCServer(grpcCtx)
+	done := make(chan os.Signal, 1)
+	signal.Notify(done, syscall.SIGINT, syscall.SIGTERM)
+
+	<-done
+	httpCancel()
+	grpcCancel()
```
Comment (eddycharly) [inline main.go:87]: is there a way to shut them down concurrently ?

**Pull request merged**